### PR TITLE
split delegate

### PR DESCRIPTION
**Pull request by askarovit**

### CODE REVIEW

File: api/controller/delegate.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import { Request, Response, NextFunction, Error } from 'express';
+import { DelegateRepository } from 'api/repository/delegate';
+import { delegateSchema as schema } from 'api/schema/delegate';
+import { Account } from 'shared/model/account';
+import { DelegateRepository as SharedDelegateRepository } from 'shared/repository/delegate';
+import { AccountPGQLRepository as SharedAccountRepository } from 'shared/repository/accountImpl';
+const constants = require('../../backlog/helpers/constants');
+
+@Controller('/api')
```
Comment (Verbalist) [inline api/controller/delegate.ts]: delegate

File: api/controller/delegate.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    public async getForgedByAccount(req: Request, res: Response) {}
+
+    @PUT('/')
+    @validate(schema.addDelegate)
+    public async addDelegate(req: Request, res: Response) {
+        const { publicKey } = req.body;
+        const account: Account = await this.sharedAccountRepo.getAccount({ publicKey });
+
+        if (!account || !account.publicKey) { /** 'Multisignature account not found'*/ }
+        if (!account.multisignatures || !account.multisignatures) {/**'Account does not have multisignatures enabled'*/}
```
Comment (Verbalist) [inline api/controller/delegate.ts]: remove all multisignatures logic

File: api/controller/delegate.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        /**
+         * Body checks
+         */
+    }
+
+    @GET('/getNextForgers')
```
Comment (Verbalist) [inline api/controller/delegate.ts]: next_forgers

File: api/controller/delegate.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+const constants = require('../../backlog/helpers/constants');
+
+@Controller('/api')
+export class DelegateController {
+    private repo = new DelegateRepository();
+    private sharedDelegateRepo = new SharedDelegateRepository();
+    private sharedAccountRepo = new SharedAccountRepository();
+
+    @GET('/count')
+    public async count(req: Request, res: Response) {
```
Comment (Verbalist) [inline api/controller/delegate.ts]: change (req, res) -> (data)

File: api/controller/delegate.ts (hunk 3 above)
Comment (Verbalist) [reply api/controller/delegate.ts]: And change all routing to unserscore

File: api/repository/delegate.ts [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+    insertFork(fork: IForkDelegate): Promise<Responce<void>>;
+
+    getDelegatesFromPreviousRound(): Promise<Responce<string[]>>;
+
+    apply(trs: Transaction<Object>, block: Block, sender: Account): Promise<Responce<void>>;
+
+    undo(trs: Transaction<Object>, block: Block, sender: Account): Promise<Responce<void>>;
+
+    applyUnconfirmed(trs: Transaction<Object>, block: Block, sender: Account): Responce<void>;
```
Comment (Verbalist) [inline api/repository/delegate.ts]: remove  all not delegate logic

File: api/service/delegate.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+interface IResponceGetBlockSlotData {
+    time: bigint;
+    keypair: Object;
+}
+
+export interface IDelegateService {
+    getBlockSlotData(slot: number, height: number): Promise<Responce<IResponceGetBlockSlotData>>;
+
+    forge(): Promise<void>;
```
Comment (Verbalist) [inline api/service/delegate.ts]: same remove all not delegate logic

File: api/repository/delegate.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    blockHeight: number;
+    previousBlock: string;
+    cause: string;
+}
+
+export interface IDelegateRepository {
+    count(): Promise<Response<{ count: number }>>;
+
+    search(q: string, limit: number, sortField: string, sortMethod: string):
+        Promise<Response<{delegates: Delegate[]}>>;
```
Comment (alex-arugon) [inline api/repository/delegate.ts]: @Verbalist should repo return not 'clear' types?

Comment (Verbalist) [reply api/repository/delegate.ts]: yes, Promise correct

**Pull request merged**